Support for compact prop in JSONFormatter

dfurtado opened the pull request:
Adds support for compact properties in the `JSONFormatter`, this will replace the default properties "level", "timestamp" and "message" by property named using its first letter prefixed by `@` symbol. For example: timestamp will be displayed as @t when `Compact` is set to true.

dfurtado committed:
Support for compact prop in JSONFormatter

Adds support for compact properties in the JSONFormatter, this will
replace the default properties "level", "timestamp" and "message" by
property named using its first letter prefixed by @ symbol. For example:
timestamp will be displayed as @t when Compact is set to true.

dfurtado committed:
Fixed property name

dfurtado committed:
fixed unit tests

Added test case for compact json format.

dfurtado committed:
code formatting

dfurtado committed:
Set stdout to false while running tests

To avoid output logs to stdout when running the tests.

dfurtado committed:
Fixed concurrency issue

ConsulFirmin commented on scopedLogger.go:
```diff
@@ -15,6 +16,15 @@ type ScopedLogger struct {
 	fields    Fields
 	logger    *Logger
 	formatter Formatter
+	mux       sync.Mutex
```
mux is an unfortunate name for a mutex. Instead mux implies a MUltipleXer of some sort. Not go specific.

dfurtado replied:
Ok, I thought that `mux` is a "standand" people use, since I have seen in other packages, and even in the golang playground examples: https://tour.golang.org/concurrency/9 but I can definitely change.

dfurtado replied:
Changed to `mutex`

ConsulFirmin commented on wlog.go:
```diff
@@ -142,7 +152,7 @@ func (l *Logger) WithScope(fields Fields) *ScopedLogger {
 		scopeFields[k] = v
 	}
-	return &ScopedLogger{scopeFields, l, l.formatter}
+	return &ScopedLogger{scopeFields, l, l.formatter, sync.Mutex{}}
```
sync.Mutex needs no initialization. It's redundant. Please remove

dfurtado replied:
Fixed

ConsulFirmin reviewed (commented):
Left a couple of comments inline. Don't know if you can see them though.

One was regarding naming a mutex to mux (which is confusing. Mux stands for MUltipleXer)
The other was about sync.Mutex that needs no initialization

dfurtado committed:
Changes after the review

- changed the mutex is not initialized when creating a instance of
ScopedLogger.
- renamed "mux" to "mutex"

dfurtado commented:
Thanks for the code review @ConsulFirmin  I have pushed all the things that needed to be fixed.
